### PR TITLE
feat(apigatewayv2): allow websockets routes to return response to client

### PR DESCRIPTION
**Pull request by emilrowland**
This PR allows WebSocket routes to return a response to the client.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by emilrowland** [comment]
I had to add an integration test to aws-apigatewayv2-integrations as I couldn't use a Mock integration in aws-apigatewayv2.
This is due to dependency conflicts.

File: packages/@aws-cdk/aws-apigatewayv2/lib/websocket/route.ts [hunk 1]
```diff
@@ -37,6 +37,13 @@ export interface WebSocketRouteOptions {
    * @default - No Authorizer
    */
   readonly authorizer?: IWebSocketRouteAuthorizer;
+
+  /**
+   * Should the route send a response to the client
+   * @default false
+   */
+  readonly shouldReturnResponse?: boolean;
```
Comment (Naumel) [inline packages/@aws-cdk/aws-apigatewayv2/lib/websocket/route.ts:45]: 0.02$: is the `should` not superfluous?

Comment (emilrowland) [reply packages/@aws-cdk/aws-apigatewayv2/lib/websocket/route.ts:45]: do you think it should be named only `returnResponse`?

File: packages/@aws-cdk/aws-apigatewayv2/lib/websocket/route.ts [hunk 2]
```diff
@@ -37,6 +37,13 @@ export interface WebSocketRouteOptions {
    * @default - No Authorizer
    */
   readonly authorizer?: IWebSocketRouteAuthorizer;
+
+  /**
+   * Should the route send a response to the client
+   * @default false
+   */
+  readonly returnResponse?: boolean;
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-apigatewayv2/lib/websocket/route.ts:45]: Given that the way this is being used when it is true, I'm not sure that boolean is the right value here. You're adding `$default` as the routeResponseSelectionExpression creating a whole object from this . Can you help me understand why a boolean here is the right user experience?

File: packages/@aws-cdk/aws-apigatewayv2-integrations/test/websocket/integ.route-response.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env node
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-apigatewayv2-integrations/test/websocket/integ.route-response.ts:1]: This line not needed

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Thank you for your contribution! Please see my comments inline.

File: packages/@aws-cdk/aws-apigatewayv2/lib/websocket/route.ts (hunk 2 above)
Comment (emilrowland) [reply packages/@aws-cdk/aws-apigatewayv2/lib/websocket/route.ts:45]: Today the only valid value o the routeResponseSelectionExpression is `$default` or `undefined`. So to not overcomplicate it by passing a string I abstracted it to be a boolean instead.


See the quote below from https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api-selection-expressions.html
> Evaluation of the route response selection expression produces a route response key. Eventually, this key will be used to choose from one of the [RouteResponses](https://docs.aws.amazon.com/apigatewayv2/latest/api-reference/apis-apiid-routes-routeid-routeresponses.html) associated with the API. However, currently only the $default key is supported.

But if we want to be more future proof we could change the attribute to be the actual `routeResponseSelectionExpression` and pass it a string.

**Comment by TheRealAmazonKendra** [comment]
@Merfigyio update

**Review by comcalvi — APPROVED** [review]
lgtm, leaving `do-not-merge` so @TheRealAmazonKendra can take a look

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**